geniza/corpus/management/commands/import_data.py

```diff
@@ -345,3 +394,173 @@ def log_creation(self, *objects):
+        If not found, tries to use first/last initials for lookup. If all else
+        fails, use the generic team account (TEAM_USERNAME).
+        """
+
+        # check the cache first
+        user = self.user_lookup.get(name)
+        if user:
+            logger.debug(
+                f"using cached user {user} for {name} on PGPID {pgpid}")
+            return user
+
+        # person with given name(s) and last name – case-insensitive lookup
+        if " " in name:
+            given_names, last_name = [sname.strip(punctuation) for sname in
+                                      name.rsplit(" ", 1)]
+            try:
+                user = User.objects.get(first_name__iexact=given_names,
+                                        last_name__iexact=last_name)
+            except User.DoesNotExist:
+                pass
+
+        # initials; use first & last to do lookup
+        elif name:
+            name = name.strip(punctuation)
+            first_i, last_i = name[0], name[-1]
+            try:
+                user = User.objects.get(first_name__startswith=first_i,
+                                        last_name__startswith=last_i)
+            except User.DoesNotExist:
+                pass
+
+        # if we didn't get anyone through either method, warn and use team user
+        if not user:
+            logger.warning(
+                f"couldn't find user {name} on PGPID {pgpid}; using {self.team_user}")
+            return self.team_user
+
+        # otherwise add to the cache using requested name and return the user
+        logger.debug(f"found user {user} for {name} on PGPID {pgpid}")
+        self.user_lookup[name] = user
+        return user
+
+    def get_edit_history(self, input_by, date_entered, pgpid=None):
+        """Parse spreadsheet "input by" and "date entered" columns to
+        reconstruct the edit history of a single document.
+
+        Output is a list of dict to pass to log_edit_history. Each event has
+        a type (django log entry action flag), associated user, and date.
+
+        This method is designed to output a list of events that matches the
+        logic of the spreadsheet and is easy to reason about (chronologically
+        ordered).
+        """
+
+        # split both fields by semicolon delimiter & remove whitespace
+        all_input = map(str.strip, input_by.split(";"))
+        all_dates = map(str.strip, date_entered.split(";"))
+
+        # try to map every "input by" listing to a user account
+        users = []
+        for input_by in all_input:
+
+            # for coauthored events, add both users to a tuple – otherwise
+            # it's a one-element tuple with the single user
+            if " and " in input_by:
+                first, second = input_by.split(" and ")
+                users.append((self.get_user(first, pgpid),
+                              self.get_user(second, pgpid)))
+            else:
+                users.append((self.get_user(input_by, pgpid),))
```

Comment:
comment is out of date now (tuple)

this should be a one-liner — I think something like this:
```users.extend([self.get_users(u, pgpid) for u in input_by.split(" and ")])```

(you don't have to check for " and " in the string— you can just split on it and get a single-element list if it isn't present)